x-pack/plugins/enterprise_search/public/applications/app_search/components/credentials/credentials_logic.ts

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { formatApiName } from '../../utils/format_api_name';
+import { ADMIN, PRIVATE } from './constants';
+
+import { HttpLogic } from '../../../shared/http';
+import { IMeta } from '../../../../../common/types';
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { IEngine } from '../../types';
+import { IApiToken, ICredentialsDetails } from './types';
+
+interface ITokenReadWrite {
+  name: 'read' | 'write';
+  checked: boolean;
+}
```

Comment:
Totally forgot to mention this too, but this one's super optional so feel free to ignore or do it another PR if you like it :) This might be a good candidate for moving to `credentials/types`, but it's also used right here in this file, so whatever works for you!

Reply:
Good call. I will put this one up in my next PR.